systemtest/src/main/java/io/strimzi/systemtest/resources/crd/kafkaclients/KafkaOauthClientsResource.java

```diff
@@ -10,50 +10,66 @@
-    private final String oauthTokenEndpointUri;
-    private final String userName;
-
-    public KafkaOauthClientsResource(
-        String producerName, String consumerName, String bootstrapServer, String topicName, int messageCount,
-        String additionalConfig, String consumerGroup, String oauthClientId, String oauthClientSecret, String oauthTokenEndpointUri) {
-
-        super(producerName, consumerName, bootstrapServer, topicName, messageCount, additionalConfig, consumerGroup, 0);
-        this.oauthClientId = oauthClientId;
-        this.oauthClientSecret = oauthClientSecret;
-        this.oauthTokenEndpointUri = oauthTokenEndpointUri;
-        this.userName = oauthClientId;
+    private String oauthClientId;
+    private String oauthClientSecret;
+    private String oauthTokenEndpointUri;
+    private String userName;
+
+    public static class KafkaOauthClientsBuilder extends KafkaClientsBuilder<KafkaOauthClientsResource.KafkaOauthClientsBuilder> {
+        private String oauthClientId;
+        private String oauthClientSecret;
+        private String oauthTokenEndpointUri;
+        private String userName;
+
+        public KafkaOauthClientsBuilder withOAuthClientId(String oauthClientId) {
+            this.oauthClientId = oauthClientId;
+            return self();
+        }
+
+        public KafkaOauthClientsBuilder withOAuthClientSecret(String oauthClientSecret) {
+            this.oauthClientSecret = oauthClientSecret;
+            return self();
+        }
+
+        public KafkaOauthClientsBuilder withOAuthTokenEndpointUri(String oauthTokenEndpointUri) {
+            this.oauthTokenEndpointUri = oauthTokenEndpointUri;
+            return self();
+        }
+
+        public KafkaOauthClientsBuilder withUserName(String userName) {
+            this.userName = userName;
+            return self();
+        }
+
+        @Override
+        public KafkaOauthClientsResource build() {
+            return new KafkaOauthClientsResource(this);
+        }
+
+        @Override
+        protected KafkaOauthClientsResource.KafkaOauthClientsBuilder self() {
+            return this;
+        }
     }
 
-    // from existing client create new client with random consumer group (immutability)
-    public KafkaOauthClientsResource(KafkaOauthClientsResource kafkaOauthClientsResource) {
-        super(kafkaOauthClientsResource);
-        this.oauthClientId = kafkaOauthClientsResource.oauthClientId;
-        this.oauthClientSecret = kafkaOauthClientsResource.oauthClientSecret;
-        this.oauthTokenEndpointUri = kafkaOauthClientsResource.oauthTokenEndpointUri;
-        this.userName = kafkaOauthClientsResource.userName;
-
+    private KafkaOauthClientsResource(KafkaOauthClientsResource.KafkaOauthClientsBuilder builder) {
+        super(builder);
+        oauthClientId = builder.oauthClientId;
+        oauthClientSecret = builder.oauthClientSecret;
+        oauthTokenEndpointUri = builder.oauthTokenEndpointUri;
```

Comment:
Here I would  add some control  that when we use oauth client these variables should not be empty or null :)

Reply:
Yes, I wanted to add it there, thanks :)